ibis/tests/expr/test_window_frames.py

```diff
@@ -0,0 +1,481 @@
+
+    expr = grouped[t, diff.name('diff')]
+    expr2 = grouped.mutate(diff=diff)
+    expr3 = grouped.mutate([diff.name('diff')])
+
+    window_expr = (t.d - t.d.lag().over(w)).name('diff')
+    expected = t.projection([t, window_expr])
+
+    assert expr.equals(expected)
+    assert expr.equals(expr2)
+    assert expr.equals(expr3)
+
+
+def test_window_analysis_combine_preserves_existing_window():
+    t = ibis.table(
+        [('one', 'string'), ('two', 'double'), ('three', 'int32')],
+        name='my_data',
+    )
+    w = ibis.cumulative_window(order_by=t.one)
+    mut = t.group_by(t.three).mutate(four=t.two.sum().over(w))
+
+    assert mut.op().selections[1].arg.frame.start is None
+
+
+def test_window_analysis_auto_windowize_bug():
+    # GH #544
+    t = ibis.table(
+        name='airlines', schema={"arrdelay": "int32", "dest": "string", "year": "int32"}
+    )
+
+    def metric(x):
+        return x.arrdelay.mean().name('avg_delay')
+
+    annual_delay = (
+        t[t.dest.isin(['JFK', 'SFO'])].group_by(['dest', 'year']).aggregate(metric)
+    )
+    what = annual_delay.group_by('dest')
+    enriched = what.mutate(grand_avg=annual_delay.avg_delay.mean())
+
+    expr = (
+        annual_delay.avg_delay.mean()
+        .name('grand_avg')
+        .over(ibis.window(group_by=annual_delay.dest))
+    )
+    expected = annual_delay[annual_delay, expr]
+
+    assert enriched.equals(expected)
+
+
+# def test_combine_windows_with_zero_offset():
+#     w1 = ibis.window(preceding=0, following=5)
+#     w2 = ibis.window(preceding=7, following=10)
+#     w3 = w1.combine(w2)
+#     expected = ibis.window(preceding=7, following=5)
+#     assert w3 == expected
+
+#     w4 = ibis.window(preceding=3, following=0)
+#     w5 = w4.combine(w2)
+#     expected = ibis.window(preceding=3, following=10)
+#     assert w5 == expected
+
+
+# def test_combine_window_with_max_lookback():
+#     w1 = ibis.trailing_window(rows_with_max_lookback(3, ibis.interval(days=5)))
+#     w2 = ibis.trailing_window(rows_with_max_lookback(5, ibis.interval(days=7)))
+#     w3 = w1.combine(w2)
+#     expected = ibis.trailing_window(rows_with_max_lookback(3, ibis.interval(days=5)))
+#     assert_equal(w3, expected)
+
+
+# def test_combine_window_with_interval_offset(alltypes):
+#     t = alltypes
+#     w1 = ibis.trailing_range_window(preceding=ibis.interval(days=3), order_by=t.e)
+#     w2 = ibis.trailing_range_window(preceding=ibis.interval(days=4), order_by=t.f)
+#     w3 = w1.combine(w2)
+#     expected = ibis.trailing_range_window(
+#         preceding=ibis.interval(days=3), order_by=[t.e, t.f]
+#     )
+#     assert w3 == expected
+
+#     w4 = ibis.range_window(following=ibis.interval(days=5), order_by=t.e)
+#     w5 = ibis.range_window(following=ibis.interval(days=7), order_by=t.f)
+#     expected = ibis.range_window(following=ibis.interval(days=5), order_by=[t.e, t.f])
+#     w6 = w4.combine(w5)
+#     assert w6 == expected
+
+
+# def test_combine_windows(alltypes):
```

Comment:
Should any of these tests be uncommented?

Reply:
We don't have the `combine()` method anymore, so I removed them.